Agents/CarparkAgent/CarparkAPIInputAgent/src/main/java/uk/ac/cam/cares/jps/agent/carpark/TimeSeriesHandler.java

```diff
@@ -0,0 +1,279 @@
+package uk.ac.cam.cares.jps.agent.carpark;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.jooq.exception.DataAccessException;
+import uk.ac.cam.cares.jps.base.util.JSONKeyToIRIMapper;
+import uk.ac.cam.cares.jps.base.timeseries.TimeSeries;
+import uk.ac.cam.cares.jps.base.timeseries.TimeSeriesClient;
+
+import java.text.SimpleDateFormat;
+
+import uk.ac.cam.cares.jps.base.exception.JPSRuntimeException;
+
+import java.io.IOException;
+import java.time.*;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class TimeSeriesHandler {
+    public static final Logger LOGGER = LogManager.getLogger(TimeSeriesHandler.class);
+    private TimeSeriesClient<OffsetDateTime> tsclient;
+    private final List<JSONKeyToIRIMapper> mappings;
+    public static final String timeUnit = OffsetDateTime.class.getSimpleName();
+    public static final String timestampKey = "time";
+    public static final ZoneOffset ZONE_OFFSET = ZoneOffset.UTC;
+
+    /**
+     * Constructor for TimeSeriesHandler
+     * @param mappings mapping for keys to data IRIs
+     * @throws IOException
+     */
+    public TimeSeriesHandler(List<JSONKeyToIRIMapper> mappings) throws IOException {
+        this.mappings = mappings;
+    }
+
+    /**
+     * retrieve the number of timeseries from mappings
+     * @return number of timeseries
+     */
+    public int getNumberofTimeSeries() {
+        return mappings.size();
+    }
+
+    /**
+     * Set timeseries client
+     * @param tsclient timeseries cient instance
+     */
+    public void setTsClient(TimeSeriesClient<OffsetDateTime> tsclient) {
+        this.tsclient = tsclient;
+    }
+
+    /**
+     * Check whether timeseries exist and instantiate the timeseries if it does not exist
+     */
+    public void initializeTimeSeriesIfNotExist() {
```

Comment:
not critical, ok if amount of time series is not huge, recommend to use bulkInitTimeSeries